pkgdb/src/pkgdb/schemas.hh

```diff
@@ -119,7 +119,7 @@ CREATE VIEW IF NOT EXISTS v_AttrPaths AS
     UNION ALL SELECT O.id, O.parent
                    , O.attrName
                    , Parent.subtree
-                   , iif( ( Parent.system IS NULL ), O.attrName, Parent.system )
+                   , COALESCE( Parent.system, O.attrName )
```

Comment:
I'm not sure I understand what this line is doing. I get that `COALESCE` will return the first non-null value, but I can't tell what `O.attrName` is supposed to be here.

Reply:
it is functionally equivalent to the old behavior but does not rely on a SQLite3 extension ( `iif` ), and instead uses the standard SQL `COALESCE` routine.

This line inherits `system` from the parent if it is defined, otherwise if `system` is undefined, we can safely assume that the "current" `O.attrName` is a system name.

I reformatted that line to make `AS system` more visible which may help.

Reply:
I think I'm also confused as to what `O` is in that context

Reply:
If you think about this routine as a recursive function, `O` is the "current attribute" being processed.